src/main/java/com/hivemq/cli/mqtt/AbstractMqttClientExecutor.java

```diff
@@ -457,4 +460,36 @@ private MqttClient getMqttClientFromCacheOrConnect(final @NotNull Connect connec
 
         return client;
     }
+
+    private Consumer<Mqtt5Publish> buildRemainingMqtt5PublishesCallback(final @NotNull Connect connect) {
+        if (connect instanceof Subscribe) {
+            return new SubscribeMqtt5PublishCallback((Subscribe) connect);
+        }
+        else {
+            return mqtt5Publish -> {
+                if (connect.isVerbose()) {
+                    Logger.trace("received PUBLISH: {} with Message: '{}'", mqtt5Publish, new String(mqtt5Publish.getPayloadAsBytes()));
+                }
+                else if (connect.isDebug()) {
+                    Logger.debug("received PUBLISH: (Topic: {}, Message: '{}')",mqtt5Publish.getTopic(), new String(mqtt5Publish.getPayloadAsBytes()));
+                }
+            };
+        }
+    }
+
+    private Consumer<Mqtt3Publish> buildRemainingMqtt3PublishesCallback(final @NotNull Connect connect) {
+        if (connect instanceof Subscribe) {
+            return new SubscribeMqtt3PublishCallback((Subscribe) connect);
+        }
+        else {
+            return mqtt3Publish -> {
+                if (connect.isVerbose()) {
+                    Logger.trace("received PUBLISH: {}", new String(mqtt3Publish.getPayloadAsBytes()));
```

Comment:
`Logger.trace("received PUBLISH: {} with Messge: '{}'", mqtt3Publish, new String(mqtt3Publish.getPayloadAsBytes()));`

Reply:
we should use 
new String(mqtt3Publish.etPayloadAsBytes(), StandardCharsets.UTF_8) 
wenn create login output

Reply:
agreed 👍